tools/mpremote/mpremote/commands.py

```diff
@@ -174,7 +174,7 @@ def do_edit(state, args):
             os.close(dest_fd)
             state.pyb.fs_touch(src)
             state.pyb.fs_get(src, dest, progress_callback=show_progress_bar)
-            if os.system("$EDITOR '%s'" % (dest,)) == 0:
+            if os.system('%s "%s"' % (os.getenv("EDITOR"), dest)) == 0:
```

Comment:
Actually, should the first %s also be in quotes?

Reply:
No, the editor should not be in quotes only the path argument.

Reply:
```
$ ls -la /home/jimmo/bin/space\ space/vim 
lrwxrwxrwx 1 jimmo jimmo 12 Oct 15 00:18 '/home/jimmo/bin/space space/vim' -> /usr/bin/vim
$ env EDITOR="/home/jimmo/bin/space space/vim" ./tools/mpremote/mpremote.py edit main.py
edit :main.py
$ env EDITOR="/home/jimmo/bin/space space/vim" ./tools/mpremote/mpremote.py edit main.py
edit :main.py
sh: line 1: /home/jimmo/bin/space: No such file or directory
```

If I add the quotes, then it works.

(Note: this was broken before too with `$EDITOR`, but we should fix it while we're here)

Reply:
If you have spaces in the path I would propose to add extra quotes to the env variable itself.

On macOS I've been using the following with VSCode:
`env EDITOR="code -w" ./tools/mpremote/mpremote.py edit main.py`
And that has been working fine for me.
But if I add the quotes around the first `%s` I now have to do the following to make it work:
`env EDITOR='"code -w"' ./tools/mpremote/mpremote.py edit main.py`
Which I think is a bit counter intuitive. 

On Windows (with this fix) I've only been able to make it work without quotes around the first `%s`.
I haven't done extensive testing of different ways of escaping.
I don't have access to a windows machine until next week for further testing if needed.

Let me know what you think.

Reply:
I think we should leave it without quotes on the first `%s`.  When it doubt better to do minimal as possible.

But should the second `%s` use single quotes (like it was before this PR) or double quotes?  Does it matter?

Reply:
Really it should be using `subprocess.run` instead and then none of this would be an issue (my bad on the original PR that added this).

Reply:
I say let's get this in. So that it starts working on Windows. Then we can always refine it.

Reply:
@dpgeorge - From my testing it does not work with single quotes on Windows and double quotes does not seam to break anything on the other platforms.